Source/DafnyLanguageServer/Workspace/DiagnosticPublisher.cs

```diff
@@ -53,11 +64,17 @@ public class DiagnosticPublisher : IDiagnosticPublisher {
     }
 
     private void PublishGhostDiagnostics(DafnyDocument document) {
-      languageServer.TextDocument.SendNotification(new GhostDiagnosticsParams {
+
+      var newParams = new GhostDiagnosticsParams {
         Uri = document.Uri,
         Version = document.Version,
         Diagnostics = document.GhostDiagnostics.ToArray(),
-      });
+      };
+      if (previouslyPublishedGhostDiagnostics.TryGetValue(document.Uri, out var previousParams) && previousParams.Diagnostics.Equals(newParams.Diagnostics)) {
```

Comment:
Is tehre a `GetValueOrDefault` for concurrent dictionaries?

```suggestion
      if (previouslyPublishedGhostDiagnostics.GetValueOrDefault(document.Uri) == newParams.Diagnostics) {
```

Probably not…

Reply:
That doesn't exist